quai/filters/filter_system.go

```diff
@@ -384,6 +408,12 @@ func (es *EventSystem) handleChainEvent(filters filterIndex, ev core.ChainEvent)
 	}
 }
 
+func (es *EventSystem) handleUnlocksEvent(filters filterIndex, ev core.UnlocksEvent) {
+	for _, f := range filters[UnlocksSubscription] {
+		f.unlocks <- ev
```

Comment:
if you add a select and default here, then we should be good even in worst case, we will never block append in any shape or form